src/game_simulation/game.py

```diff
@@ -0,0 +1,119 @@
+import random
+import numpy as np
+
+# TODO that is only game simulation - delete later, so it is not refactored. Please, don't read
+
+
+class Game:
+
+    def __init__(self):
+        self.map = np.zeros(shape=(5, 5), dtype='int32')
+        self.player_numbers = np.array([])
+        self.player_positions = [[-1, -1], [-1, -1], [-1, -1]]
+
+    def add_player(self, player_number, position=None):
+        if player_number in self.player_numbers:
+            raise ValueError("Player with this number already exists.")
+
+        if not position:
+            self.add_random_position(player_number)
+        else:
+            if self.map[position[0], position[1]] == 0:
+                raise ValueError("The position is not empty.")
+            else:
+                self.map[position[0], position[1]] = player_number + 1
+                self.player_positions[player_number] = position
+
+        self.player_numbers = np.append(self.player_numbers, player_number)
+
+    def add_random_position(self, player_number):
+        while True:
+            x = random.randint(0, 4)
+            y = random.randint(0, 4)
+            if self.map[x][y] == 0:
+                self.map[x][y] = player_number + 1
+                self.player_positions[player_number] = [x, y]
+                return
+
+    def __str__(self) -> str:
+        map_in_string = ''
+        for row in range(5):
+            for column in range(5):
+                map_in_string = ' '.join([map_in_string, f'{self.map[row, column]} '])
+            map_in_string = ''.join([map_in_string, '\n'])
+        return map_in_string
+
+    def make_move(self, player_number, move):
+        if move == 'w':
+            self.go_up(player_number)
+        elif move == 's':
+            self.go_down(player_number)
+        elif move == 'a':
+            self.go_left(player_number)
+        else:
+            self.go_right(player_number)
```

Comment:
I think you could make some constants like KEY_UP, KEY_DOWN etc. because maybe it will be changed in future for arrow keys and them you will have to search for places they occured

Reply:
Everything is game.py - don't care about that, because it will be delete in final version, so I don't see a point to refactor.